common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/AccessCondition.scala

```diff
@@ -4,8 +4,8 @@ import java.time.Instant
 
 case class AccessCondition(
   status: AccessStatus,
-  terms: Option[String],
-  to: Option[Instant]
+  terms: Option[String] = None,
+  to: Option[Instant] = None
```

Comment:
I'm not a massive fan of having these in the actual internal model. It feels like being explicit in the implementing is better and easier to understand?

Why hide it? It's caught us up before, especially with options where `None` would work for either one of these. So if by accident I put `AccessCondition(status, None)` that's valid and doesn't say very much.

The extra code written I feel adds more clarity which I value over succinctness.

Saying that for tests and methods they seem fine, just not on internal models.

There's a good article here:
https://blog.ssanj.net/posts/2019-05-01-why-are-default-parameter-values-considered-bad-in-scala.html

@alicefuzier thoughts?

Reply:
There are some valid points, but due to the nature of our models (large case classes with lots of optional args) I think not having any default values would have a substantial impact on the ergonomics of using them (e.g. we can currently use constructs like `AccessCondition(status = _)` to make anonymous functions in a really clear way etc).

It would be a bit of a code smell to me if I saw `AccessCondition(status, None)`, especially as there is no context here as to what `None` references. For that reason it might be good for us to have a convention of always initialising case classes with named arguments, rather than positional ones (doing this would avoid pretty much all the problems alluded to).